### PR TITLE
test: Limit retries via the TestCase itself

### PR DESCRIPTION
**Pull request by stefwalter**
The Policy only has no state per test, the TestCase seems like
the more natural place.

Also, increase limit to five tries and clean up retry loop.

Signed-off-by: Stef Walter <stefw@redhat.com>
 * Simplified the retry loop
 * Use the TestCase to store number of tries
 * Squashed the two commits for clarity

Replaces #6000

### CODE REVIEW

File: test/common/testlib.py [hunk 1]
```diff
@@ -848,35 +855,34 @@ def check_issue(self, trace):
             traceback.print_exc()
         return number
 
-    def check_retry(self, trace):
-        #
+    def check_retry(self, trace, tries):
+        # Never try more than five times
+        if self.retryable and tries >= 5:
```
Comment (mvollmer) [inline test/common/testlib.py]: Shoudn't the condition be `not self.retryable or tries >= 5`?

File: test/common/testlib.py [hunk 2]
```diff
@@ -460,19 +460,26 @@ def run(self, result=None):
...
+        #
+        # Policy actually dictates retries.  The number here is an upper bound to
+        # prevent endless retries if Policy.check_retry is buggy.
+        max_retry_hard_limit = 10
+        for retry in range(0, max_retry_hard_limit):
             try:
-                self.currentResult = result
                 super(MachineCase, self).run(result)
             except RetryError, ex:
-                self.doCleanups()
```
Comment (mvollmer) [inline test/common/testlib.py]: Is this removal intentional?

**Comment by mvollmer** [comment]
What about `test: Really exit phantomjs on fatal errors` from #6000?

**Comment by mvollmer** [comment]
> What about test: Really exit phantomjs on fatal errors from #6000?

Ohh, that one made _everything_ fail...

File: test/common/testlib.py (hunk 2 above)
Comment (stefwalter) [inline test/common/testlib.py]: It's moved to the TapResult itself which is the main point for managing retries. The retry loop is as clean as possible ... it's unfortunate that the retry loop is here. I wish it could be elsewhere, not in MachineCase specific code. But realistically this is where it has to live for now. At least lets keep it really simple.

File: test/common/testlib.py (hunk 1 above)
Comment (stefwalter) [inline test/common/testlib.py]: Whoops. Will update

**Comment by stefwalter** [comment]
Fixed use of setattr().
